internal/server/metrichandler/update.go

```diff
@@ -0,0 +1,91 @@
+package metrichandler
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/shurikeagle/metrics-collector/internal/server/metric"
+)
+
+// POST /update/{metricType}/{metricName}/{metricValue}
+func (h *handler) updateHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Немного overengineered. Функции хендлера можно сразу описывать, как
```suggestion
	func (h *handler) updateHandler (w http.ResponseWriter, r *http.Request) {
```
А в раутер их вставлять в виде объектов, а не в виде вызова функции, которая возвращает функцию.
```
h.Post("/update/{metricType}/{metricName}/{metricValue}", h.updateHandler)
```
Не блокер, скорее оптимизация, для повышения читабельности.